graphql/collect_fields_cache_store.go

```diff
@@ -0,0 +1,76 @@
+package graphql
+
+import (
+	"hash/fnv"
+	"sync"
+	"unsafe"
+
+	"github.com/vektah/gqlparser/v2/ast"
+)
+
+// collectFieldsCacheKey is the cache key for CollectFields results.
+type collectFieldsCacheKey struct {
+	selectionData unsafe.Pointer // Pointer to the underlying SelectionSet data
+	selectionLen  int            // Length of the selection set
+	satisfiesHash uint64         // Hash of the satisfies array
```

Comment:
Using slice header pointers as keys for speed and to avoid memory pressure. The length field is not strictly necessary for functionality, but is included as a safeguard in case the slice gets sliced unexpectedly in the future.